src/ripple/shamap/impl/SHAMapSync.cpp

```diff
@@ -725,15 +738,13 @@ void SHAMap::getFetchPack (SHAMap* have, bool includeLeaves, int max,
 
                 if (next->isInner ())
                 {
-                    if (!have || !have->hasInnerNode (childID, childHash))
+                    if (! have || ! have->hasInnerNode (childID, childHash))
                         stack.push ({next, childID});
                 }
-                else if (includeLeaves && (!have || !have->hasLeafNode (next->peekItem()->getTag(), childHash)))
+                else if (! have || ! have->hasLeafNode (next->peekItem()->getTag(), childHash))
```

Comment:
The check for whether or not to include leaves is now done in getFetchPack. The visistDifferences functions calls the function for leaves and inner nodes.